plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/JavaPackageValidator.java

```diff
@@ -1,28 +1,33 @@
 package com.google.cloud.tools.eclipse.appengine.newproject;
 
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
 import org.eclipse.jdt.core.JavaConventions;
 import org.eclipse.jdt.core.JavaCore;
 
 public class JavaPackageValidator {
 
+  private static final String PLUGIN_ID = 
+      "com.google.cloud.tools.eclipse.appengine.newproject.AppEngineStandard";
+  
   /**
    * Check if a string is a legal Java package name.
    */
-  // todo return an IStatus for better error reporting
-  public static boolean validate(String packageName) {
+  public static IStatus validate(String packageName) {
+    
     if (packageName == null) {
```

Comment:
Why don't you let the `JavaConventions` do the heavy lifting and then do any mop-ups, like contains whitespace?  It should already do the package-ends-with-a-dot, and has standard (and localized) messaging